### PR TITLE
Frrist/init actor basic unit test

### PR DESCRIPTION
**Pull request by frrist**
### What
- add a testing util method for creating an actor address.
- implemented methods in mock runtime for mocking actor creation.
- add initial set of tests for the init actor:
  - Constructor
  - Actors that cannot exec the init actor should fail
  - Creating a payment channel via init actor
  - Creating a miner via init actor
  - Ensure all addresses can be resolved to an ID via ResolveAddress and address that cannot be resolved are returned.

### CODE REVIEW

File: actors/builtin/init/init_test.go [hunk 1]
```diff
@@ -1 +1,154 @@
...
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	assert "github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	_init "github.com/filecoin-project/specs-actors/actors/builtin/init"
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Nit: I'd prefer `init_` so that things like IDE autocomplete are more likely to be useful.

File: actors/builtin/init/init_test.go [hunk 2]
```diff
@@ -1 +1,154 @@
...
+	anne := tutil.NewIDAddr(t, 1001)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+
+	t.Run("abort actors that cannot call exec", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.SetActorCodeCID(anne, builtin.AccountActorCodeID)
+		// TODO test case where `ok` == false, current impl panics
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Not necessary, that would indicate an error in the runtime.

File: actors/builtin/init/init_test.go [hunk 3]
```diff
@@ -1 +1,154 @@
...
+
+	var fakeParams = runtime.CBORBytes([]byte{'D', 'E', 'A', 'D', 'B', 'E', 'E', 'F'})
+	var balance = abi.NewTokenAmount(100)
+
+	t.Run("happy path exec create payment channel", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		actor.constructAndVerify(rt)
+
+		// anne the account actor with 100 FIL in their balance is going to create a payment channel actor with 100 FIL.
```
Comment (anorth) [inline actors/builtin/init/init_test.go]:
```suggestion
		// anne execs a payment channel actor with 100 FIL.
```
Reference to anne's balance is confusing here.

File: actors/builtin/init/init_test.go [hunk 4]
```diff
@@ -1 +1,154 @@
...
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.SetBalance(balance)
+		rt.SetReceived(balance)
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Consider adding a second parameter to SetReceived, being the new balance, since the balance always needs to be updated to reflect the received amount. But leave the standalone SetBalance there too.

File: actors/builtin/init/init_test.go [hunk 5]
```diff
@@ -1 +1,154 @@
...
+		rt.SetActorCodeCID(anne, builtin.AccountActorCodeID)
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Idea: update SetCaller to also populate this value, making this call unnecessary here.

File: actors/builtin/init/init_test.go [hunk 6]
```diff
@@ -1 +1,154 @@
...
+
+		// re-org-stable address of the payment channel actor
+		uniqueAddr := tutil.NewActorAddr(t, "paych")
+		rt.SetNewActorAddress(uniqueAddr)
+
+		// next id address
+		expectedIdAddr := tutil.NewIDAddr(t, 100)
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Please test creating two actors in a row, and that they get different addresses.

File: actors/builtin/init/init_test.go [hunk 7]
```diff
@@ -1 +1,154 @@
...
+		execRet := actor.execAndVerify(rt, builtin.PaymentChannelActorCodeID, fakeParams)
+		assert.Equal(t, uniqueAddr, execRet.RobustAddress)
+		assert.Equal(t, expectedIdAddr, execRet.IDAddress)
+
+		var st _init.InitActorState
+		rt.GetState(&st)
+		actualIdAddr, err := st.ResolveAddress(rt.Store(), uniqueAddr)
+		assert.NoError(t, err)
+		assert.Equal(t, expectedIdAddr, actualIdAddr)
+
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Nit: trailing blank line

File: actors/builtin/init/init_test.go [hunk 8]
```diff
@@ -1 +1,154 @@
...
+		// should return the same address if not able to resolve
+		expUnknowAddr := tutil.NewActorAddr(t, "flurbo")
+		actualUnknownAddr, err := st.ResolveAddress(rt.Store(), expUnknowAddr)
+		assert.NoError(t, err)
+		assert.Equal(t, expUnknowAddr, actualUnknownAddr)
+
+	})
```
Comment (anorth) [inline actors/builtin/init/init_test.go]: Add a test for the constructor failing (ExpectSend returning non-zero).

File: support/mock/builder.go [hunk 9]
```diff
@@ -36,7 +36,9 @@ func NewBuilder(ctx context.Context, receiver addr.Address) *RuntimeBuilder {
 		expectValidateCallerAddr: nil,
 		expectValidateCallerType: nil,
 
-		expectSends: make([]*expectedMessage, 0),
+		expectSends:   make([]*expectedMessage, 0),
+		actorCodeCIDs: make(map[addr.Address]cid.Cid),
+		newActorAddr:  addr.Undef,
```
Comment (anorth) [inline support/mock/builder.go]: Move these last two up with the other fake values (rather than mock expectations), line ~33

File: support/mock/mockrt.go [hunk 10]
```diff
@@ -47,6 +49,12 @@ type Runtime struct {
 	expectValidateCallerAddr []addr.Address
 	expectValidateCallerType []cid.Cid
 	expectSends              []*expectedMessage
+	expectCreateActor        *expectCreateActorPair
+}
+
+type expectCreateActorPair struct {
```
Comment (anorth) [inline support/mock/mockrt.go]: `expectedCreateActor`

File: support/mock/builder.go [hunk 11]
```diff
@@ -36,7 +36,9 @@ func NewBuilder(ctx context.Context, receiver addr.Address) *RuntimeBuilder {
 		expectValidateCallerAddr: nil,
 		expectValidateCallerType: nil,
```
Comment (anorth) [inline support/mock/builder.go]: `expectCreateActor: nil` for completeness
